nodeadm/internal/aws/ecr/ecr.go

```diff
@@ -19,12 +19,15 @@ func GetAuthorizationToken(awsRegion string) (string, error) {
 		return "", err
 	}
 	ecrClient := ecr.NewFromConfig(awsConfig)
-	token, err := ecrClient.GetAuthorizationToken(context.Background(), &ecr.GetAuthorizationTokenInput{})
-	if err != nil {
-		return "", err
+
+	for i := 0; i < 3; i++ {
```

Comment:
could use our internal `util.RetryExponentialBackoff` for this

Reply:
We couldn't use it directly here as the `util.RetryExponentialBackoff` function could accept method which will return error only. In this case, we need to return both token string and error, as the token is required to pull sandbox image.

I'm not sure if it is worthy to create another retry method like `func RetryExponentialBackoff(attempts int, initial time.Duration, f func() (string, error)) error ` which function in param could return both string and err.

Reply:
correct me if im wrong, but you can just use a closure for the token, something like 
```go
var token *ecr.GetAuthorizationTokenOutput
util.RetryExponentialBackoff(2, time.Second, func() error {
  token, err = ecrClient.GetAuthorizationToken(context.Background(), &ecr.GetAuthorizationTokenInput{})
  return err
})
```

Reply:
This might not work as it expected for following reasons. 
* First is because `func` passed to method `RetryExponentialBackoff` here is a function declare. And the token `var token *ecr.GetAuthorizationTokenOutput` declared outside of this func declare is not actually passed to this func.
```
func() error {
  token, err = ecrClient.GetAuthorizationToken(context.Background(), &ecr.GetAuthorizationTokenInput{})
  return err
}
```
* Where this func actually run is in`util.RetryExponentialBackoff`. So if we want to implement like passing a point param to this func(), 

  * The `token1` declared here need to be passed to `RetryExponentialBackoff` method. And also we need to define the func declare here to accept another `token2` in param
  ```
  var token1 *ecr.GetAuthorizationTokenOutput
  err := util.RetryExponentialBackoff(token1, 3, 2*time.Second, func(token2 *ecr.GetAuthorizationTokenOutput) error {
		token2, err = ecrClient.GetAuthorizationToken(context.Background(), &ecr.GetAuthorizationTokenInput{})
		return err
  })
  ```

  * And in `RetryExponentialBackoff` method, we can pass `token1` param to `f(token1)` where the func is actually run.
  ```
  func RetryExponentialBackoff(token1 *ecr.GetAuthorizationTokenOutput, attempts int, initial time.Duration, f func(*ecr.GetAuthorizationTokenOutput) error) error {
	var err error
	wait := initial
	for i := 0; i < attempts; i++ {
		if err = f(token1); err == nil {
			return nil
		}
		time.Sleep(wait)
		wait *= 2
	}
	return err
  }
  ```

But no matter how, it will require us to create another new `RetryExponentialBackoff` method in util.go